aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -762,6 +843,32 @@ private int createMultipartUploadRecords(String bucket, String key, File file, O
         return dbUtil.bulkInsertTransferRecords(valuesArray);
     }
 
+    private File writeInputStreamToFile(InputStream inputStream) throws IOException {
+        if (inputStream == null) {
+            throw new IllegalArgumentException("Invalid inputStream: " + inputStream);
+        }
+
+        // Saves the data as a file in the temporary directory
+        File file = File.createTempFile("aws-s3", ".tmp");
+        if (!file.isFile()) {
+            throw new IOException("Error creating a temporary file.");
+        }
+        try (OutputStream outStream = new FileOutputStream(file)) {
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+            // Keep reading until reaches the end of the stream
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                outStream.write(buffer, 0, bytesRead);
+                outStream.flush();
+            }
+            inputStream.close();
```

Comment:
As written, this `inputStream` will only be closed if any of the code above does not throw an exception.    Normally to solve this, you would add a `finally` block after the `catch`, and close it there, so that it gets closed no matter what.  

That said, [whoever opens a stream is typically responsible for closing it](https://stackoverflow.com/a/23925742/282279).  Since the customer will be opening the `InputStream`, I think we should leave it up to the customer to close it.

Reply:
Agree. Since `inputStream` is an input to this method, we should not close it on our will. So I will delete line 864.